Add initial test stubs for Remote Execution

JacobCallahan opened the pull request:
The Remote Execution (Management) feature is currently confined to UI.
These initial test stubs are the current manual testing cases.

oshtaier commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,238 @@
+from robottelo.test import UITestCase
+
+
+class TestRemoteExecution(UITestCase):
```
`TestRemoteExecution` -> `RemoteExecutionTestCase`

oshtaier commented:
Please refer to https://github.com/SatelliteQE/robottelo/issues/3083 for our naming conventions.
All test cases should use words `positive` or `negative` after word `test_`

oshtaier commented:
Don't need to use extra line in the end of docstring. Change
```
 +        @Status: Manual
 +
 +        """
```
on
```
 +        @Status: Manual
 +        """
```

oshtaier commented:
Please classify all your test cases using corresponding `tier` decorators as in all our tests

oshtaier commented:
Please, don't use numbers in test names (e.g. `test_re_create_job_template_1`) and provide meaningful name instead

oshtaier commented:
Pending comments

JacobCallahan commented:
Thanks for the great input, @oshtaier!

sthirugn commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,238 @@
+
+    @stubbed()
+    @run_only_on('sat')
+    def test_re_create_job_template_1(self):
```
remove `re_` part from all test cases as all the test cases in this module belong to re - remote execution.

sthirugn replied:
Also tests need to have `positive`, `negative` text added in the test case names whenever possible.

JacobCallahan commented:
Addressed all previous comments. I'm not 100% on the tiers, but these should be about right.

oshtaier commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,227 @@
+from robottelo.test import UITestCase
+
+
+class TestRemoteExecution(UITestCase):
```
Not sure what is changed, but I still see `TestRemoteExecution` title when it should be like I mentioned in my previous comment:
`TestRemoteExecution` -> `RemoteExecutionTestCase`

oshtaier commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,227 @@
+
+    @stubbed()
+    @tier1
+    def test_positive_create_job_input_template(self):
```
It seems more like `test_positive_create_job_template_input`, but not sure on 100%

oshtaier commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,227 @@
+
+    @stubbed()
+    @tier1
+    def test_negative_create_job_template(self):
```
`test_negative_create_job_template_with_same_name` is more applicable here.
I suggest also adding test case that targets invalid names for job template (e.g. too long strings used for name).

oshtaier commented:
Can we add few test cases for update or delete procedures for mentioned `job templates`? I didn't see UI for that functionality, so please sorry in advance for incorrect suggestions

oshtaier commented:
Also, please add corresponding entry to robottelo/docs/api/tests.foreman.ui.rst

sghai commented:
@JacobCallahan : hey..good to see tests around remote execution.  Are you planning to cover more scenarios around this later ?

rplevka commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,227 @@
+        @Steps:
+
+        1. Create a valid job template
+        2. Enter a name that has already been used
```
You might want to add a step of pre-creating the 1st job template first. You can use the @Setup section of the docstring for such steps

sthirugn replied:
:+1:

rplevka commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,227 @@
+
+    @stubbed()
+    @tier1
+    def test_positive_preview_verify_file(self):
```
not sure about this one but shouldn't this be a negative scenario?

rplevka replied:
negative?

oshtaier replied:
+1

sthirugn commented on tests/foreman/ui/test_remoteexecution.py:
```diff
@@ -0,0 +1,246 @@
+
+        @Assert: Verify the job was not immediately ran
+        @Assert: Verify the job was succesfully ran after the designated time
+
```
Only one `@Assert` section allowed.  So probably you can number these different assertions under `@Assert` section like how you did for `@Steps`

oshtaier replied:
+1

sthirugn commented:
Pending @rplevka's comment about using `@setup` section for pre-requisites.

oshtaier commented on docs/api/tests.foreman.ui.rst:
```diff
@@ -198,6 +198,11 @@
 .. automodule:: tests.foreman.ui.test_puppetclass
 
+:mod:`tests.foreman.ui.test_remoteexecution`
+---------------------------------------
```
few more symbols to avoid new warning
```
:mod:`tests.foreman.ui.test_remoteexecution`
--------------------------------------------
```

oshtaier commented:
ACK
pending my small comment and @rplevka ones

rplevka commented:
ACK
pending comments :)

JacobCallahan commented:
I believe I have addressed all previous comments. Thanks for the great input, guys. A number of things have changed since my last PR!! 

I also cleaned up a few other lines, per what @oshtaier recommended.

sthirugn commented:
ACK

rplevka commented:
gg,
ACK

omaciel commented:
@JacobCallahan :100:
